mint-auditor/src/db/models/audited_mint.rs

```diff
@@ -0,0 +1,696 @@
+        let mut deposit = create_gnosis_safe_deposit(100, &mut rng);
+        insert_gnosis_deposit(&mut deposit, &conn);
+
+        // Create MintTxs with a mismatching amount.
+        let (_mint_config_tx, signers) = create_mint_config_tx_and_signers(token_id1, &mut rng);
+        let mut mint_tx = create_mint_tx(token_id1, &signers, deposit.amount() + 1, &mut rng);
+
+        mint_tx.prefix.nonce = hex::decode(&deposit.expected_mc_mint_tx_nonce_hex()).unwrap();
+
+        // Insert the MintTx to the database, and check that the mismatch is
+        // detected.
+        MintTx::insert_from_core_mint_tx(0, None, &mint_tx, &conn).unwrap();
+        assert!(matches!(
+            AuditedMint::attempt_match_deposit_with_mint(&deposit, config, &conn),
+            Err(Error::DepositAndMintMismatch(_))
+        ));
+
+        // Check that nothing was written to the `audited_mints` table
+        assert_audited_mints_table_is_empty(&conn);
+
+        // Mismatch counter was incremented
+        assert_eq!(
+            Counters::get(&conn)
+                .unwrap()
+                .num_mismatching_mints_and_deposits(),
+            1
+        );
+    }
+
+    #[test_with_logger]
+    fn test_attempt_match_deposit_with_mint_unsaved_object(logger: Logger) {
+        let config = &test_gnosis_config().safes[0];
+        let mut rng = mc_util_test_helper::get_seeded_rng();
+        let test_db_context = TestDbContext::default();
+        let mint_auditor_db = test_db_context.get_db_instance(logger.clone());
+        let conn = mint_auditor_db.get_conn().unwrap();
+
+        let deposit = create_gnosis_safe_deposit(100, &mut rng);
+
+        assert!(matches!(
+            AuditedMint::attempt_match_deposit_with_mint(&deposit, config, &conn),
+            Err(Error::ObjectNotSaved)
+        ));
+
+        // Check that nothing was written to the `audited_mints` table
+        assert_audited_mints_table_is_empty(&conn);
+    }
+
+    #[test_with_logger]
+    fn test_attempt_match_deposit_with_mint_mismatched_safe_addr(logger: Logger) {
+        let mut config = test_gnosis_config().safes[0].clone();
+        let mut rng = mc_util_test_helper::get_seeded_rng();
+        let test_db_context = TestDbContext::default();
+        let mint_auditor_db = test_db_context.get_db_instance(logger.clone());
+        let conn = mint_auditor_db.get_conn().unwrap();
+
+        let mut deposit = create_gnosis_safe_deposit(100, &mut rng);
+        insert_gnosis_deposit(&mut deposit, &conn);
+        insert_mint_tx_from_deposit(&deposit, &conn, &mut rng);
+
+        config.safe_addr = EthAddr::from_str("0x0000000000000000000000000000000000000000").unwrap();
+        assert!(matches!(
+            AuditedMint::attempt_match_deposit_with_mint(&deposit, &config, &conn),
+            Err(Error::Other(_))
+        ));
+
+        // Check that nothing was written to the `audited_mints` table
+        assert_audited_mints_table_is_empty(&conn);
```

Comment:
If the logic and test case above gets updated
```suggestion
        assert_audited_mints_table_is_empty(&conn);

        assert_eq!(
            Counters::get(&conn)
                .unwrap()
                .num_unexpected_errors_matching_deposits_to_mints(),
            1
        );
```